test/x509aux.c

```diff
@@ -173,59 +150,31 @@ static int test_certs(BIO *fp)
     }
 
     /* Some other PEM read error */
-    print_errors();
     return 0;
 }
 
-int main(int argc, char *argv[])
+int test_main(int argc, char *argv[])
 {
     BIO *bio_err;
-    const char *p;
     int ret = 1;
 
-    progname = argv[0];
-    if (argc < 2) {
-        test_usage();
-        EXIT(ret);
-    }
+    if (argc < 2)
+        TEST_error("usage: %s certfile...", argv[0]);
 
     bio_err = BIO_new_fp(stderr, BIO_NOCLOSE | BIO_FP_TEXT);
 
-    p = getenv("OPENSSL_DEBUG_MEMORY");
-    if (p != NULL && strcmp(p, "on") == 0)
-        CRYPTO_set_mem_debug(1);
-    CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_ON);
-
-    argc--;
-    argv++;
-
-    while (argc >= 1) {
+    for (argv++; *argv; argv++) {
         BIO *f = BIO_new_file(*argv, "r");
         int ok;
 
-        if (f == NULL) {
-            fprintf(stderr, "%s: Error opening cert file: '%s': %s\n",
-                    progname, *argv, strerror(errno));
-            EXIT(ret);
-        }
-        ret = !(ok = test_certs(f));
+        TEST_check(f != NULL);
+        ok = test_certs(f);
```

Comment:
This runs the test directly rather than using the test runner logic (which has implications on the reporting I assume). Wouldn't it be better to use ADD_ALL_TESTS()? Granted you would have to find some way of passing the BIO or filename to the test itself (elsewhere we have used global variables to pass command line info from "main" to the tests).